feat(wkt): Support jiff From conversions for Duration and Timestamp

fiadliel opened the pull request:
jiff is a promising time handling crate based on the design of the Temporal JavaScript API. It is planned for a long-term 1.0 version to be released later in the year.

This commit adds support for converting to and from `jiff::SignedDuration` and `jiff::Timestamp` for version 0.2.

To ease the transition between version 0.2 and 1.0, I've named the feature `jiff-0_2`, so that support for multiple versions could coexist. But if preferred, I can rename the feature to just `jiff`.

Fixes #1153

coryan commented:
Generally this looks good.  I assume the motivation for naming the feature `jiff-0_2` is to provide (eventually) `jiff-0_3` and `jiff` for a future 1.0 release?  I would like to avoid an implicit commitment to support multiple versions.  Though I guess we can always remove supported versions while our own version is 0.x (we would need to bump the `0.x` to 0.${x + 1}`, of course).
Thoughts?

PS: not asking you to change anything yet, just trying to understand motivations and tradeoffs.

fiadliel commented:
Updated the feature name as `jiff`.

Also updated the `Timestamp` tests to verify that the jiff and wkt values both represent the same point in time (otherwise, it proves that the first value in the test case was converted to the other, but not that the test case values were wisely chosen).

coryan commented:
I think we are going to wait until `jiff` gains more traction before taking a dependency on it.  It looks very promising, and once it reaches 1.0 I think the case is easier.  For now, we want to be conservative on adding extra deps to our public API (even if protected by feature flags).

I hope you do not feel like you wasted a lot of time on this.  We are still refining our thinking about how to manage deps for Rust.

fiadliel commented:
No hard feelings here - it's fair. Also given the beta nature of both these libraries _at this point in time_, it's not going to inconvenience a lot of users.

I'll close this PR for now, but you can use it as reference code in the future if you want.
